Enterprise-CMCS/macpro-mdct-mfp
# WP - Autosave and hydration - EntityDetailsOverlay page type

### Description


### What this PR covers
- Updating fields in entity details overlay page types now autosaves and hydrates correctly after a save of any kind
- Autosave fix for choicelist now allows for nested fields to autosave + hydrate
- Updating value in I. Define Initiative page reflects correct value in IV. Close Out page
- refactored all of state management 
- cleaned up unused code
- Creating reports should now correctly set target populations

### What this PR does not cover/yet to be worked on
- disabled close out button + return to dashboard button [CMDCT-201](https://jiraent.cms.gov/browse/CMDCT-201)
- Pressing "Yes, close out initiative" or "Save and return" takes user back to the initiative dashboard [CMDCT-204](https://jiraent.cms.gov/browse/CMDCT-204)
- Brainstorm naming conventions for page types
- `Other, specify` topic in Funding Sources

### Related ticket(s)


CMDCT-194

---
### How to test

- Create a new initiative in the WP 
- Go to the I. Define initiative page and fill out fields 
- Confirm there was an autosave on blur (tab off and check the network tab for an api call) 
- Leave page and return, confirm correct hydration of values
- Go to IV. Close out page and confirm that the disabled field has correct projected date value from I. Define Initiative page
- Like the last page, confirm autosave and hydration
- Confirm that statusing, creating multiple initiatives, overlaymodal page type are all working as expected, and that any data hydration is as expected



### Important updates




---
### Author checklist


---


_convert to a different template: [test → val](?expand=1&template=test-to-val-deployment.md)_ | _[val → prod](?expand=1&template=val-to-prod-deployment.md)_
